base/strings/string.jl

```diff
@@ -340,8 +343,15 @@ julia> repeat('A', 3)
 "AAA"
 ```
 """
-repeat(c::AbstractChar, r::Integer) = repeat(Char(c), r) # fallback
-function repeat(c::Char, r::Integer)
+# don't assume effects for general integers since we cannot know their implementation
+@assume_effects :foldable function repeat(c::Char, r::BitInteger)
```

Comment:
Changing this to `BitInteger` broke [these docs](https://github.com/JuliaLang/julia/blob/6d678fec0ad0c78e80f357aac1b1e99ff0ff47ca/doc/src/base/strings.md#L14)